src/app/resources/iis_scripts/local_http.ps1

```diff
@@ -36,12 +56,33 @@ $uri = $uriBuilder.ToString()
 
 try {
     $httpMsg = New-Object System.Net.Http.HttpRequestMessage -ArgumentList $httpMethod, $uri
-    if ($reqObj.Body) {
+    if ($reqObj._body) {
+        $requestContent = New-Object System.Net.Http.StringContent ([string] $reqObj._body)
         $httpMsg.Content = $requestContent
-        $requestContent = New-Object System.Net.Http.StringContent ([string] $reqObj.Body)
     }
     foreach ($prop in $reqObj.headers | Get-Member -MemberType NoteProperty | Select-Object -ExpandProperty Name) {
-        $httpMsg.Headers.Add($prop, $reqObj.headers.$prop)
+        $headerValue = $reqObj.headers.$prop
+        if (!$httpMsg.Headers.TryAddWithoutValidation($prop, $headerValue)) {
+            $headerFixed = $false
+            if ($httpMsg.Content) {
+                if ($prop -like "content-type") {
+                    $httpMsg.Content.Headers.ContentType = New-Object System.Net.Http.Headers.MediaTypeHeaderValue $headerValue
+                    $headerFixed = $true
+                } elseif ($prop -like "content-range") {
+                    $tokens = $prop.Split("/-")
+                    $from = [int]::Parse(($tokens[0].Trim() -split " ")[-1])
+                    $to = [int]::Parse($tokens[1].Trim())
+                    $length = [int]::Parse($tokens[2].Trim())
+                    $httpMsg.Content.Headers.ContentRange = New-Object System.Net.Http.Headers.ContentRangeHeaderValue -ArgumentList $from, $to,  $length
+                }
+                ## possibly add more misplaced headers
+            }
+            if ($headerFixed) {
+                LogVerbose "Replaced header ${prop}: ${headerValue}"
+            } else {
+                LogVerbose "Failed to add header ${prop}: ${headerValue}"
+            }

```

Comment:
I think we should also throw exception here to make it easy to find the unfixed header issue.